propolis/src/hw/pci/bridge.rs

```diff
@@ -110,41 +100,218 @@ impl Bridge {
+
+                // These registers appear in type-0 PCI headers, but not bridge
+                // headers.
+                StdCfgReg::MaxLatency
+                | StdCfgReg::MinGrant
+                | StdCfgReg::CardbusPtr => {
+                    panic!("Unexpected register type {:?}", id);
                 }
-                CfgReg::Common(id) => self.pci_state.cfg_std_read(id, ro, ctx),
-                _ => self.inner.lock().unwrap().cfg_std_read(id, ro, ctx),
             },
-            RWOp::Write(wo) => match id {
-                CfgReg::Common(id) => {
-                    self.pci_state.cfg_std_write(self, id, wo, ctx)
+            BridgeReg::PrimaryBus => {
+                let guard = self.inner.lock().unwrap();
+                ro.write_u8(guard.primary_bus.get());
+            }
+            BridgeReg::SecondaryBus => {
+                let guard = self.inner.lock().unwrap();
+                ro.write_u8(guard.secondary_bus.get());
+            }
+            BridgeReg::SubordinateBus => {
+                let guard = self.inner.lock().unwrap();
+                ro.write_u8(guard.subordinate_bus.get());
+            }
+            BridgeReg::SecondaryLatencyTimer => ro.write_u8(0),
+            BridgeReg::IoBase | BridgeReg::IoLimit => ro.write_u8(0),
+            BridgeReg::SecondaryStatus => ro.write_u16(BRIDGE_SECONDARY_STATUS),
+            BridgeReg::MemoryBase => {
+                let guard = self.inner.lock().unwrap();
+                ro.write_u16(guard.memory_base & BRIDGE_MEMORY_REG_MASK);
+            }
+            BridgeReg::MemoryLimit => {
+                let guard = self.inner.lock().unwrap();
+                ro.write_u16(guard.memory_limit & BRIDGE_MEMORY_REG_MASK);
+            }
+            BridgeReg::PrefetchableMemoryBase
+            | BridgeReg::PrefetchableMemoryLimit => ro.write_u16(0),
+            BridgeReg::PrefetchableMemoryBaseUpper
+            | BridgeReg::PrefetchableMemoryLimitUpper => ro.write_u32(0),
+            BridgeReg::IoBaseUpper | BridgeReg::IoLimitUpper => ro.write_u16(0),
+            BridgeReg::BridgeControl => ro.write_u16(0),
+        }
+    }
+
+    fn cfg_std_write(&self, id: &BridgeReg, wo: &mut WriteOp, _ctx: &DispCtx) {
+        match id {
+            BridgeReg::Common(id) => match id {
+                // Ignore writes to read-only standard registers.
+                StdCfgReg::VendorId
+                | StdCfgReg::DeviceId
+                | StdCfgReg::Class
+                | StdCfgReg::Subclass
+                | StdCfgReg::SubVendorId
+                | StdCfgReg::SubDeviceId
+                | StdCfgReg::ProgIf
+                | StdCfgReg::RevisionId
+                | StdCfgReg::HeaderType
+                | StdCfgReg::CapPtr
+                | StdCfgReg::CacheLineSize
+                | StdCfgReg::LatencyTimer
+                | StdCfgReg::Bist
+                | StdCfgReg::Reserved => {}
+                StdCfgReg::Command => {
```

Comment:
style nit: a newline between `Command` and the ignored writes case would make it easier to tell them apart visually

Reply:
Done.